python/ql/src/semmle/python/regex.qll

```diff
@@ -133,7 +141,19 @@ abstract class RegexString extends Expr {
             end in [start + 2 .. start + 4] and
             exists(this.getText().substring(start + 1, end).toInt())
             or
-            this.getChar(start + 1) != "x" and end = start + 2
+            this.getChar(start + 1) = "U" and end = start + 10
+            or
+            this.getChar(start + 1) = "N" and
+            this.getChar(start + 2) = "{" and
+            this.getChar(end - 1) = "}" and
+            end > start and
+            not exists(int i |
+              i > start + 2 and
+              i < end - 1 and
+              this.getChar(i) = "}"
+            )
```

Comment:
It feels like a predicate like this should be pulled out. In fact, I might prefer all the cases to be either named predicates or have a comment like `octal` on them. Such a refactor would spill over into existing code, though.

Reply:
I agree, this would be more readable if the Unicode name aliases were handled as a separate predicate. For some of the less complicated bits, a comment will suffice, I think.